### DISCUSSION TITLE
Creating an API for postgres database by using postgREST, CURL -Reg

### DISCUSSION THREAD

**Discussion by sureshbellamkonda96** [PostgREST/postgrest#3657] [Q&A] [answered]
Hi,

Using the postgREST option, I intend to construct an API for my Postgres database. I produced a configuration file with my server and database details (localhost, port, etc.) and installed postgREST binaries for that purpose. I then began using postgREST, and I was able to observe as seen below:

Starting PostgREST 12.2.1 (******)...
Listening on port ****
 Attempting to connect to the database...
: Successfully connected to PostgreSQL 15.4, compiled by Visual C++ build 1914, 64-bit
Listening for notifications on the "pgrst" channel
Config reloaded
 Schema cache queried in 15.6 milliseconds
 Schema cache loaded 50 Relations, 0 Relationships, 280 Functions, 0 Domain Representations, 4 Media Type Handlers

After postgREST had begun, I installed CURL for web requests, opened a command window, and typed the following : 
             curl http://localhost:****/table name       then i got error like below you can see

     ERROR:   {"code":"PGRST302","details":null,"hint":null,"message":"Anonymous access is disabled"}

 So kindly understand the environment and hope you drag me out from this issue.

### Environment

* PostgreSQL version: 15
* PostgREST version: v12.2.1
* Operating system: windows

### Description of issue

After installing postgREST and CURL, I encountered difficulties in developing an API for my Postgres database.  The error is:
         {"code":"PGRST302","details":null,"hint":null,"message":"Anonymous access is disabled"}
Could you please help on this. Thanks in advance.
Regards,
Suresh.

**Comment by wolfgangwalther** [comment, marked as answer]
Please ask support questions like this in the "Discussions" section and not create new Issues for them. I moved the issue over to the Q & A discussion section.

---

```
{"code":"PGRST302","details":null,"hint":null,"message":"Anonymous access is disabled"}
```

The message tells you that you can not access the api without authenticating.

You also have a PGRST302 error code, which you can look up here: https://docs.postgrest.org/en/v12/references/errors.html#group-3-jwt

This will point you to the right configuration option there.

You should also consider finishing the two tutorials we have:
- https://docs.postgrest.org/en/v12/tutorials/tut0.html
- https://docs.postgrest.org/en/v12/tutorials/tut1.html

Those deal with exactly those things.
